docker/zap-baseline.py

```diff
@@ -271,23 +330,119 @@ def main(argv):
-                      '-addonupdate',
-                      '-addoninstall', 'pscanrulesBeta']  # In case we're running in the stable container
-
-            if zap_alpha:
-                params.append('-addoninstall')
-                params.append('pscanrulesAlpha')
-
-            add_zap_options(params, zap_options)
+        if use_af and af_supported:
+            print('Using the Automation Framework')
+
+            # Generate the yaml file
+            yaml_file = '/zap/zap.yaml'
+            summary_file = '/zap/zap_out.json'
+            
+            with open(yaml_file, 'w') as yf:
+
+                # Add the top level to the scope for backwards compatibility
+                top_levels = [ target ]
+                if target.count('/') > 2:
+                    # The url can include a valid path, but always reset to spider the host (backwards compatibility)
+                    t2 = target[0:target.index('/', 8)+1]
+                    if not t2 == target:
+                        target = t2
+                        top_levels.append(target)
+               
+                yaml.dump(get_af_env(top_levels, debug), yf)
+                
+                addons = ['pscanrulesBeta']
+                if zap_alpha:
+                    addons.append('pscanrulesAlpha')
+                    
+                jobs = [
+                        get_af_addons(addons, []),
+                        get_af_pscan_config(),
+                        get_af_spider(target, mins)                        ]
+                
+                if ajax:
+                    jobs.append(get_af_spiderAjax(target, mins))
+                
+                jobs.append(get_af_pscan_wait(timeout))
+                jobs.append(get_af_output_summary(('Short', 'Long')[detailed_output], summary_file))
+                
+                if report_html:
+                    jobs.append(get_af_report('traditional-html', base_dir, report_html, 'ZAP Scanning Report', ''))
+            
+                if report_md:
+                    jobs.append(get_af_report('traditional-md', base_dir, report_md, 'ZAP Scanning Report', ''))
+            
+                if report_xml:
+                    jobs.append(get_af_report('traditional-xml', base_dir, report_xml, 'ZAP Scanning Report', ''))
+            
+                if report_json:
+                    jobs.append(get_af_report('traditional-json', base_dir, report_json, 'ZAP Scanning Report', ''))
+            
+                yaml.dump({'jobs': jobs}, yf)
+                
+                if os.path.exists('/zap/wrk'):
+                    # Write the yaml file to the mapped directory, if there is one
+                    copyfile(yaml_file, '/zap/wrk/zap.yaml')
+ 
+            # Run ZAP inline with the yaml file
+            try:
+                params = ['-autorun', yaml_file]
+    
+                add_zap_options(params, zap_options)
+    
+                out = run_zap_inline(port, params)
```

Comment:
Could also use the existing `start_zap()` function and read from `zap.out`?

Reply:
No, because thats starts ZAP in the background and we wont know when to stop it. For the AF its better to run inline.